server/stt/stt_planning_ml.py

```diff
@@ -124,7 +139,253 @@ async def get_coverage_details(
+                coverage["planning"]["scheduled"] = self.datetime(
+                    workstartdate_elt.text
+                )
+            except (ValueError, TypeError):
+                logger.warning(
+                    f"Failed to parse workstartdate: {workstartdate_elt.text}"
+                )
+
+        # Parse Finnish text fields
+        self.parse_finnish_text_fields(planning_elt, coverage)
+
+    def add_field_to_coverage(
+        self, coverage: Dict[str, Any], field_name: str, value: str
+    ):
+        """Add a field to coverage.planning.fields list"""
+        if value:
+            coverage["planning"]["fields"].append({"field": field_name, "value": value})
+
+    def parse_coverage_status(self, subject_elt: Element, coverage: Dict[str, Any]):
+        """Parse coverage status from sttworkstatus subject using CV from DB"""
+        qcode = subject_elt.get("qcode", "")
+
+        # Map STT internal workstatus to Superdesk coverage status
+        status_mapping = {
+            "sttworkstatus:1": "ncostat:int",
+            "sttworkstatus:2": "ncostat:int",
+            "sttworkstatus:3": "ncostat:int",
+            "sttworkstatus:4": "ncostat:notint",
+            "sttworkstatus:5": "ncostat:notdec",
+        }
+
+        mapped_status = status_mapping.get(qcode)
+        if mapped_status:
+            # Get coverage status from vocabulary
+            coverage_status = self.get_coverage_status_from_vocabulary(mapped_status)
+            if coverage_status:
+                coverage["news_coverage_status"] = coverage_status
+
+    def get_coverage_status_from_vocabulary(
+        self, qcode: str
+    ) -> Optional[Dict[str, Any]]:
+        """Get coverage status from newscoveragestatus vocabulary"""
+        try:
+            vocab_service = get_resource_service("vocabularies")
+            coverage_status_vocab = vocab_service.find_one(
+                req=None, _id="newscoveragestatus"
+            )
+
+            if coverage_status_vocab and "items" in coverage_status_vocab:
+                for item in coverage_status_vocab["items"]:
+                    if item.get("qcode") == qcode and item.get("is_active", True):
+                        return {
+                            "qcode": item["qcode"],
+                            "name": item.get("name", ""),
+                            "label": item.get("label", ""),  # Use label from vocabulary
+                        }
+        except Exception as e:
+            logger.warning(
+                f"Failed to get coverage status from vocabulary for {qcode}: {e}"
+            )
+
+        return None
+
+    def parse_picture_type(self, subject_elt: Element, coverage: Dict[str, Any]):
+        """Parse picture type from sttimagetypename subject"""
+        qcode = subject_elt.get("qcode", "")
+
+        # Map STT numeric image type codes to new CV qcodes
+        image_type_mapping = {
+            "sttimagetypename:20": "sttimage:20",  # Kuvaaja paikalla
```

Comment:
Where is this mapping coming from, I couldn't find it?

Reply:
This mapping comes from the updated sttimagetype vocabulary defined in `server/data/vocabularies.json` and `STT_Imagetype.json.`